src/main/java/entities/Topic.java

```diff
@@ -59,12 +88,52 @@ public boolean isInQueue(Member member) {
 
     /**
      * Remove and return the next Member in the queue.
-     * @return The Member at the front of the queue
+     * @return The new mentee
      */
     public Member getNextFromQueue() {
         return queue.remove();
     }
 
+    /**
+     * Create a new mentoring room for this topic.
+     * @param mentee The mentee for this room
+     * @return The new Room
+     */
+    public Room createRoom(Member mentee) {
+        // get the lowest unused number
+        int number = 1;
+        for (int i = 1; i <= rooms.size() + 1; i++) {
+            if (rooms.get(Room.makeName(name, i)) == null) {
+                number = i;
+                break;
+            }
+        }
```

Comment:
Depending on how strongly you want to keep to the "one task per method" paradigm, finding the lowest unused room number should be its own method.

Reply:
In fact, I would urge against reusing room numbers. Benefit of not: 1) you can easily see how many mentoring request you've had. 2) it's not confusing if someone got the same room number twice (edge case). 3) easy to implement, a static int in the Room class that is incremented on every call to new(). In the case the you decide to keep reusing room numbers, I think a better way to do it would be to have a static queue that contains all of the reusable numbers. Flow would go something like this:
1) new room, check if queue is empty
2) if yes, use rooms.size() + 1, or a static int ROOM_COUNT
3) if no, pop int from queue and use
4) room deleted, add number to queue

The time complexity of using a queue is much better, O(1) I believe, while the worst case of the current method is O(n) or worse (addressing collisions in the HashMap). Given that there could be a monster amount of rooms, O(n) could bog down the system.

Reply:
Also you avoid doing a boat load of string concatenation

Reply:
Good points. I suppose that if we're providing direct links to the channels, it doesn't matter how long the channel names are. I'll switch this to using sequential numbering.